src/Tools/dotnet-watch/test/GlobbingAppTests.cs

```diff
@@ -28,71 +31,71 @@ public GlobbingAppTests(ITestOutputHelper logger)
         public async Task ChangeCompiledFile(bool usePollingWatcher)
         {
             _app.UsePollingWatcher = usePollingWatcher;
-            await _app.StartWatcherAsync();
+            await _app.StartWatcherAsync().TimeoutAfter(DefaultTimeout);
```

Comment:
nit: most of these already have timeouts internally, I was suggesting finding the ones that didn't already have timeouts (if any) and adding it to those functions so all tests benefit.

Also, looks like GetCompiledAppDefinedTypes already has a timeout https://github.com/dotnet/aspnetcore/blob/c5490d3b6976f7ee49b616a98b0fbbe0393c6089/src/Tools/dotnet-watch/test/GlobbingAppTests.cs#L143

Reply:
I don't mind doubling up these already-quite-long timeouts given there's a demonstrated hang. I plan on merging this without changes. I didn't double up the timeout where it was obvious there was a timeout already, but I don't want to have to inspect every async method call in this class just in case there's a nested timeout in there somewhere.

I do wonder if even timeouts will catch the issue though given what I saw in the dump. Given there was nothing related to the hanging test on the heap, that implies the awaited Task got GCd and the test continuations along with it. In that case, the test would hang but our Task timeout extension methods would have no hope of catching it unless we and some finalizer magic.